### ISSUE TITLE
Don't require a ring argument in factor_poly()

### ISSUE THREAD

**Issue by dkahle** [coneill-math/m2r#176]
This behavior is inconsistent with what you get with `ideal()`, for example, where the last ring used is defaulted.

```r
(R <- ring("x", coefring = "QQ"))
# M2 Ring: QQ[x], grevlex order
factor_poly("x^2 - 1", ring = R)
# $factor
# x  -  1
# x  +  1
# 
# $power
# [1] 1 1
# 

ring("x", coefring = "QQ")
# M2 Ring: QQ[x], grevlex order
factor_poly("x^2 - 1")
#  Error in inherits(x, "m2_pointer") : 
#   argument "ring" is missing, with no default 
```

**Comment by coneill-math** [comment]
This should probably extend to other functions as well.  Just like in Macaulay2, the default should be to use the last created ring, unless specified otherwise.

**Comment by dkahle** [comment]
Closed with #184.
